### ISSUE TITLE
Sik-Ho Tang | Review -- Split-Brain Autoencoders: Unsupervised Learning by Cross-Channel Prediction.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#126]
Sik-Ho Tang. [Review — Split-Brain Autoencoders: Unsupervised Learning by Cross-Channel Prediction](https://medium.com/geekculture/review-split-brain-autoencoders-unsupervised-learning-by-cross-channel-prediction-cebeba644a3c).

**Comment by NorbertZheng** [comment]
## Overview
Split-Brain Auto for Self-Supervised Learning, Outperforms Jigsaw Puzzles, [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [ALI/BiGAN](https://medium.com/nerd-for-tech/review-bigan-adversarial-feature-learning-gan-535eb76be2ca), [L³-Net](https://sh-tsang.medium.com/review-look-listen-and-learn-self-supervised-learning-ff89a7dee980), [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997), etc.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/95f688ec-7ce8-43a9-9bde-9baaf34929f0)
Proposed Split-Brain Auto (Bottom) vs Traditional Autoencoder, e.g. Stacked Denoising Autoencoder (Top).

In this paper, Split-Brain Autoencoders: Unsupervised Learning by Cross-Channel Prediction, (Split-Brain Auto), by Berkeley AI Research (BAIR) Laboratory, University of California, is reviewed. In this paper:
 - A network is split into two sub-networks, each is trained to perform a difficult task:
    - **predicting one subset of the data channels from another.**
 - By forcing the network to solve cross-channel prediction tasks, feature learning is achieved without using any labels.

This is a paper in 2017 CVPR with over 400 citations.

**Comment by NorbertZheng** [comment]
## Split-Brain Autoencoders (Split-Brain Auto)
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/d8762467-c987-48fa-ad30-8531d980be57)
Split-Brain Autoencoders applied to various domains.

### Cross-Channel Encoders
 - First, input data $X$ is divided into $X_{1}$ and $X_{2}$.
 - Then, $X_{1}$ goes through network $F_{1}$ to predict $X_{2}$:
   ![image](https://github.com/NorbertZheng/read-papers/assets/37235023/b34c9da8-0baf-46df-a97d-5d6881411943)

By performing this pretext task of predicting $X_{2}$ from $X_{1}$, we hope to achieve a representation $F(X_{1})$ which contains high-level abstractions or semantics.

Similar for $F_{2}$ that $X_{2}$ goes through network $F_{2}$ to predict $X_{1}$.
 - Left: For Lab color space, $X_{1}$ can be L, which is luminance information, and $X_{2}$ can be ab, which are color information.
 - Right: For RGB-D image, $X_{1}$ can be RGB values, and $X_{2}$ can be D, which is depth information.

$l_{2}$-loss can be used to train the regression loss:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a9e9350c-629d-4229-a362-32206085fd1a)

 - **It is found that the cross-entropy loss is more effective than $l_{2}$ loss for the graphics task of automatic colorization than regression:**
   ![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a96d0373-31fa-414d-9a9b-a8bdcad18436)

**Comment by NorbertZheng** [comment]
Interesting!!! Cross-entropy loss is better than $l_{2}$ loss.

**Comment by NorbertZheng** [comment]
### Split-Brain Autoencoders as Aggregated Cross-Channel Encoders
Multiple cross-channel encoders, $F_{1}$, $F_{2}$, on opposite prediction problems, with loss functions $L_{1}$, $L_{2}$, respectively:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/e872c819-cb9c-4993-baa1-7b3c369724d9)

Example split-brain autoencoders in the image and RGB-D domains are shown in the above figure (a) and (b), respectively.

 - **By concatenating the representations layer-wise, $F^{l} = \\{F_{1}^{l}, F_{2}^{l}\\}$, a representation $F$ is achieved which is pre-trained on full input tensor $X$.**

If F is a CNN of a desired fixed size, e.g., [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160), we can design the subnetworks F1, F2 by splitting each layer of the network F in half, along the channel dimension.

 - **The network is modified to be fully convolutionally and trained for a pixel-prediction task.**

**Comment by NorbertZheng** [comment]
### Alternative Aggregation Technique
One alternative, as a baseline: The same representation $F$ can be trained to perform both mappings simultaneously:
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/31b880c4-29b4-4418-9653-8d912d70812f)

Or even considering the full input tensor $X$.
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/37a285a8-0fd8-40ec-b01e-b5cab0817a98)

 - **However, it is found that the proposed Split-Brain Auto (Section 1.2) outperforms the above two alternatives (Section 1.3).**

**Comment by NorbertZheng** [comment]
Step-by-step training is better!!! Reduce the difficulty of training task!!!

**Comment by NorbertZheng** [comment]
## Experimental Results
### ImageNet
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/81448696-4ac8-48aa-82aa-9782ae7ae525)
Task Generalization on ImageNet Classification.

Model:
 - The proposed split-brain autoencoder architecture learns the unsupervised representations on large-scale image data from ImageNet.
 - Lab color space is used to train the split-brain autoencoder.
 - All weights are frozen and feature maps spatially resized to be ~9000 dimensions.
 - All methods use [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) variants.

Dataset:
 - The 1.3M (i.e. 1'300'000) ImageNet dataset without labels is used for training, except for ImageNet-labels.

To be brief, different autoencoder variants are tried.

Split-Brain Auto (cl, cl), cl means using classification loss, outperforms all variants and all self-supervised learning approaches such as Jigsaw Puzzles [30], [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff) [7], [Ali [8]/BiGAN](https://medium.com/nerd-for-tech/review-bigan-adversarial-feature-learning-gan-535eb76be2ca), [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997) [34] and Colorization [47].

**Comment by NorbertZheng** [comment]
### Places
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ec54c2d4-7069-410c-912a-9a3c6fb2f35c)
Dataset & Task Generalization on Places Classification.

A different task (Places) than the pretraining tasks (ImageNet).

Similar results are obtained for Places Classification, it outperforms such as Jigsaw Puzzles [30], [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff) [7], [L³-Net](https://sh-tsang.medium.com/review-look-listen-and-learn-self-supervised-learning-ff89a7dee980) [45], [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997) [34] and Colorization [47].

**Comment by NorbertZheng** [comment]
### PASCAL VOC
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/921b2c22-310d-4bc9-ac9a-399b4830c16f)
Task and Dataset Generalization on PASCAL VOC.

To further test generalization, classification, detection and segmentation performance is evaluated on PASCAL VOC.

The proposed method, Split-Brain Auto (cl, cl), achieves state-of-the-art performance on almost all established self-supervision benchmarks.

**Comment by NorbertZheng** [comment]
There are still other results in the paper. If interested, please feel free to read the paper. Hope I can write a story about Jigsaw Puzzles in the coming future.

**Comment by NorbertZheng** [comment]
## Reference
 - [2017 CVPR] [Split-Brain Auto] [Split-Brain Autoencoders: Unsupervised Learning by Cross-Channel Prediction](https://arxiv.org/abs/1611.09842).
